packages/smart-contracts/test/contracts/ERC20TransferableReceivable.test.ts

```diff
@@ -0,0 +1,283 @@
+import '@nomiclabs/hardhat-ethers';
+import { BytesLike, Signer } from 'ethers';
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import {
+  TestERC20__factory,
+  TestERC20,
+  ERC20TransferableReceivable__factory,
+  ERC20TransferableReceivable,
+  ERC20FeeProxy,
+  ERC20FeeProxy__factory,
+} from '../../src/types';
+import { BigNumber as BN } from 'ethers';
+
+const BASE_DECIMAL = BN.from(10).pow(BN.from(18));
+
+describe('contract: ERC20TransferableReceivable', () => {
+  let deployer: Signer;
+  let user1: Signer, user1Addr: string;
+  let user2: Signer, user2Addr: string;
+  let user3: Signer, user3Addr: string;
+
+  let testToken: TestERC20, receivable: ERC20TransferableReceivable, erc20FeeProxy: ERC20FeeProxy;
+
+  before(async function () {
+    [deployer, user1, user2, user3] = await ethers.getSigners();
+    user1Addr = await user1.getAddress();
+    user2Addr = await user2.getAddress();
+    user3Addr = await user3.getAddress();
+  });
+
+  beforeEach(async function () {
+    testToken = await new TestERC20__factory(deployer).deploy(BN.from(1000000).mul(BASE_DECIMAL));
+    erc20FeeProxy = await new ERC20FeeProxy__factory(deployer).deploy();
+    receivable = await new ERC20TransferableReceivable__factory(deployer).deploy(
+      'Request Network Transferable Receivable',
+      'tREC',
+      erc20FeeProxy.address,
+    );
+
+    await testToken.approve(receivable.address, ethers.constants.MaxUint256);
+  });
+
+  async function verifyReceivables(userAddr: string, receivableIds: any) {
+    const ids = await receivable.getTokenIds(userAddr);
+    expect(ids.toString()).to.equals(receivableIds.toString());
+  }
+
+  describe('mint', async function () {
```

Comment:
Can we add a test where we show that the tokenURI is stored and can be read? All the current tests make tokenURI empty string and never checks it.

Reply:
Nevermind, I'm wrong. This functionality is tested.

Reply:
will do